### DISCUSSION TITLE
Cannot generate symbols.zip in Android studio

### DISCUSSION THREAD

**Discussion by VeerammalV** [community/community#143881] [Programming Help]
### Body

I'm trying to deploy a project in play store. I installed CMake and NDK (side by side) and using NDK= "FULL" in my build gradle. When I build my signed APK, I got merged_native_libs file. But debug symbol.zip file is not generated. I don't know where I made mistake and I couldn't find any reason why it's not generated. Do anyone know how to generate debug symbol.zip file?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by mastertanzeel** [comment]
Hi @VeerammalV,

To generate the `debug-symbols.zip` file in an Android project, here are some steps and checks to ensure everything is set up correctly:

### 1. **Confirm the Correct Build Configuration in `build.gradle`**
   Ensure the following configurations are in your app-level `build.gradle` file:

   ```gradle
   android {
       ...
       buildTypes {
           release {
               minifyEnabled true // Enable Proguard/R8 for release builds
               shrinkResources true
               ndk {
                   debugSymbolLevel 'FULL' // Sets the debug symbol level to full
               }
           }
       }
       ...
   }
   ```

   Setting `debugSymbolLevel` to `'FULL'` instructs the build to include all debug symbols in the generated `.so` files.

### 2. **Confirm Your NDK and CMake Installation**
   - **NDK Version**: Make sure your NDK version is compatible with the version of your Android Gradle plugin. Some issues can occur with incompatible NDK versions. It’s typically recommended to use the default NDK version suggested by Android Studio for compatibility.
   - **CMake**: Confirm CMake is installed via **File > Settings > Appearance & Behavior > System Settings > Android SDK > SDK Tools** and check **CMake** and **NDK (Side by Side)**.

### 3. **Generate the `debug-symbols.zip` File Manually**
   If it’s still not generated automatically, you can generate it manually:

   - Run the following Gradle command from your project’s root directory to build the release APK with symbols:

     ```bash
     ./gradlew assembleRelease
     ```

   - After building, navigate to the `app/build/outputs` directory. You should see a `mapping` or `symbols` folder (e.g., `app/build/outputs/symbols/release/`), where you can manually zip the symbols if they aren’t auto-zipped.

### 4. **Use Android Studio’s Built-in Export Tool**
   You can also use Android Studio’s **Generate Signed Bundle / APK** feature, which should generate the necessary debug symbols automatically.

   1. Go to **Build > Generate Signed Bundle / APK**.
   2. Select **APK**.
   3. Complete the wizard, ensuring **Release** is selected and symbols are enabled in the `build.gradle`.

### 5. **Enable Debug Symbols in Project Structure (If Available)**
   If you're using Android Studio's **Project Structure** settings, navigate to **File > Project Structure > Modules > [Your App Module] > Build Variants**. Under the **Build Variants** section, confirm that debug symbols are set to "full" or "minimal."

---

If after these steps you still don’t see `debug-symbols.zip`, it may be helpful to clear your project cache by running:

```bash
./gradlew clean
```

Then re-run the build to see if the file is generated. Let me know if any of these steps help!

**Reply by VeerammalV** [reply]
From your answer, Im clear that my setup for NDK and CMake was correct.. I've cleaned and rebuilt the project many times, invalidated cache and restarted many times.  It didn't work.. I'm gonna try the 3rd one you suggested and let you know the result.. Thank you so much..

**Reply by mastertanzeel** [reply]
You're very welcome! It sounds like you've been thorough in your troubleshooting so far, so trying the manual `debug-symbols.zip` generation route might indeed be the key here. Sometimes the manual approach can reveal more about what's missing or misconfigured, especially if Android Studio's automatic processes are getting stuck.

Give it a go, and please feel free to reach out if you encounter anything unexpected. Best of luck, and I’m here to help if needed!
